.github/pull_request_template.md

```diff
@@ -6,6 +6,19 @@
 
 <!-- List the affected areas (e.g wallet, browser, etc..) -->
 
+### Developer Checklist
+
+#### Tests & QA
+- [ ] I have created user stories for this feature/rework
+- [ ] I have written unit tests for status-go and QML changes
+- [ ] I have created a storybook page for all UI components and views
+- [ ] I have tested this PR on both storybook and the actual client (for UI changes)
+- [ ] I have followed the architecture guidelines: [Architecture guidelines](../architecture-guide.md)
+- [ ] For bug fixes, I have written a new unit test case to prevent reintroduction
+- [ ] I have performed side impact testing for changes in shared components
+- [ ] A reviewer has Tested the PR during review
```

Comment:
these two might be better in a reviewer checklist, not sure if thats a thing though.